openbas-api/src/main/java/io/openbas/rest/inject/service/InjectService.java

```diff
@@ -61,9 +66,91 @@ public class InjectService {
   private final InjectStatusRepository injectStatusRepository;
   private final InjectMapper injectMapper;
   private final MethodSecurityExpressionHandler methodSecurityExpressionHandler;
+  private final UserService userService;
+  private final InjectorContractService injectorContractService;
+  private final TagRuleService tagRuleService;
+  private final TagService tagService;
+  private final DocumentService documentService;
 
   @Resource protected ObjectMapper mapper;
 
+  public Inject createInject(
+      @Nullable final Exercise exercise,
+      @Nullable final Scenario scenario,
+      @NotNull final InjectInput input) {
+    if (exercise == null & scenario == null) {
```

Comment:
Sanity check need to be done on rest part.
Should have one exercise or one scenario, not both, at least one.